addons/binding/org.openhab.binding.roku/src/main/java/org/openhab/binding/roku/internal/handler/RokuHandler.java

```diff
@@ -0,0 +1,125 @@
+import org.openhab.binding.roku.internal.RokuState;
+import org.openhab.binding.roku.internal.protocol.RokuCommands;
+import org.openhab.binding.roku.internal.protocol.RokuCommunication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RokuHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jarod Peters - Initial contribution
+ * @auther Shawn Wilsher - Overhaul of channels and properties
+ */
+public class RokuHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(RokuHandler.class);
+
+    public RokuHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        if (command.toFullString().equals("REFRESH")) {
+            updateData();
+        } else {
+            RokuCommands rokuDevice = new RokuCommands(ipAddress, port);
+            try {
+                rokuDevice.generateAction(channelUID);
+            } catch (IOException e) {
+                logger.error("IOException occurred when attempting action", e);
+            }
+            updateState(channelUID, OnOffType.OFF);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing RokuDevice handler '{}'", getThing().getUID());
+        Number refreshInterval = (Number) this.getConfig().get(REFRESH_INTERVAL);
+        scheduler.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                updateData();
+            }
+        }, 0, refreshInterval.longValue(), TimeUnit.SECONDS);
+        logger.debug("Polling job scheduled to run every {} sec. for '{}'", refreshInterval, getThing().getUID());
+    }
+
+    private synchronized void updateData() {
+        logger.info("Update Roku data '{}'", getThing().getUID());
+        String ipAddress = (String) this.getConfig().get(IP_ADDRESS);
+        Number port = (Number) this.getConfig().get(PORT);
+        RokuState state = new RokuState(new RokuCommunication(ipAddress, port));
+        try {
+            state.updateDeviceInformation();
+        } catch (IOException e) {
+            logger.debug("Roku device '{}' but is not communicating", ipAddress, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+        }
+
+        updatePropertiesFromState(state);
+
+        updateState(CHANNEL_ACTIVE, state.getActive());
+        updateState(CHANNEL_HEADPHONES, new StringType(state.getHeadphonesConnected()));
+        updateState(CHANNEL_ICON, state.getActiveImage());
+        /**
+         * Future Functionality
+         * updateState(CHANNEL_APPBROWSER, state.getApplicationMenu());
+         */
+        if (!state.getPowerMode().equals("PowerOn")) {
```

Comment:
Position literals first in String comparisons - that way if the String is null you won't get a NullPointerException, it'll just return false. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)